span.go

```diff
@@ -490,6 +495,40 @@ func (s *Span) setExitSpanDestinationService() {
 	})
 }
 
+func (s *Span) setExitSpanServiceTarget() {
+	fallbackType := s.Subtype
+	if fallbackType == "" {
+		fallbackType = s.Type
+	}
+
+	// Service target fields explicitly provided.
+	if s.Context.setServiceTargetCalled {
+		// if the user calls SetServiceTarget with a non-empty name, but empty type,
+		// we'll use the specified name and infer the type
+		if s.Context.serviceTarget.Type == "" && s.Context.serviceTarget.Name != "" {
+			s.Context.SetServiceTarget(ServiceTargetSpanContext{
```

Comment:
Should we have a unit test for this case, where SetServiceTarget is called with only a name?

Reply:
Good point!